File: 2017/08.py

import math, copy, re

REGISTERS = dict()

def compare(reg, n, comp):
	if comp == '==':
		return REGISTERS[reg] == n
	elif comp == '!=':
		return REGISTERS[reg] != n
	elif comp == '<':
		return REGISTERS[reg] < n
	elif comp == '<=':
		return REGISTERS[reg] <= n
	elif comp == '>':
		return REGISTERS[reg] > n
	elif comp == '>=':
		return REGISTERS[reg] >= n
	return 

def part_1(data):
	for cmd in data:
		[reg1, op, n, _, reg2, comp, c] = cmd.split()
		if reg1 not in REGISTERS.keys():
			REGISTERS[reg1] = 0
		if reg2 not in REGISTERS.keys():
			REGISTERS[reg2] = 0

		if compare(reg2, int(c), comp):
			if op == 'inc':
				REGISTERS[reg1] += int(n)
			elif op == 'dec':
				REGISTERS[reg1] -= int(n)
	print(max(REGISTERS.values()))

	return

def part_2(data):
	maxi = 0
	for cmd in data:
		[reg1, op, n, _, reg2, comp, c] = cmd.split()
		if reg1 not in REGISTERS.keys():
			REGISTERS[reg1] = 0
		if reg2 not in REGISTERS.keys():
			REGISTERS[reg2] = 0

		if compare(reg2, int(c), comp):
			if op == 'inc':
				REGISTERS[reg1] += int(n)
			elif op == 'dec':
				REGISTERS[reg1] -= int(n)
		if REGISTERS[reg1] > maxi:
			maxi = REGISTERS[reg1]
	print(maxi)
	return 


if __name__ == '__main__':
	with open('8_input') as f:
		data = f.read()
		data = data.split('\n')
		# data = list(map(int, data.split()))


	# part_1(copy.deepcopy(data))
	part_2(copy.deepcopy(data))
	

File: 2017/08.js

const fs = require('fs');
const { exec } = require("child_process");
const { ALPHA_L, ALPHA_U, D4, D6, D8, MOD } = require("../lib");
const { joseph, findCycle } = require("../lib");
const { cache } = require("../lib");
const { bin, float, hex, int, num, oct } = require("../lib");
const { range, drange, trange, iter, diter, titer } = require("../lib");
const { copy,	entries, in_, inBB, keys, sort, values } = require("../lib");
const { Counter, DD, empty, FastQueue, PriorityQueue, Queue, set, Stack } = require("../lib");
const { areaInt, circumference, manDist, shoelace } = require("../lib");
const { ord, chr, count, debug, disp, disp3, crt, gcd, lcm, modPow, modPowBig, modInv, mod, prod, prodBig, randint, sum, sumBig, transpose } = require("../lib");
const { digits, ints,	floats,	singles,	words,	lines,	table,	groups,	getGroups,	groupsWith,	parse,	parseLine, scanf } = require ('../lib');
const { min, max, random, abs, ceil, floor, log, log10, log2, round, sign, sin, cos, tan, asin, acos, atan, atan2, sqrt, PI } = Math;
const { isSuperset, or, and, xor, sub } = set;
const { getExecStr } = require("../lib/post");
const { combinations, combinations_with_replacement, next_permutation, product } = require("../lib");

const CMP = {
	'>': (a, b) => a > b,
	'<': (a, b) => a < b,
	'>=': (a, b) => a >= b,
	'<=': (a, b) => a <= b,
	'==': (a, b) => a == b,
	'!=': (a, b) => a != b,
}

function part1(data) {

	let res = 0;
	data = scanf(data, "%w %w %d if %w %$ %d"/*yxr inc 119 if nev != 6*/)
	const D = new DD();
	iter(data, ([reg, dir, n, r, cond, m]) => {
		if (CMP[cond](D[r], m)) {
			switch (dir) {
			case 'inc':
				D[reg] += n;
				break;
			case 'dec':
				D[reg] -= n;
				break
			default:
				console.assert(false);
			}
		}
	});
	res = -MOD;
	iter(keys(D), k => {
		res = max(res, D[k]);
	})

	debug(res);
	if (res) exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART1");
	return;
}

function part2(data) {

	let res = -MOD;
	data = scanf(data, "%w %w %d if %w %$ %d"/*yxr inc 119 if nev != 6*/)
	const D = new DD();
	iter(data, ([reg, dir, n, r, cond, m]) => {
		if (CMP[cond](D[r], m)) {
			switch (dir) {
			case 'inc':
				D[reg] += n;
				break;
			case 'dec':
				D[reg] -= n;
				break
			default:
				console.assert(false);
			}
		}
		res = max(res, D[reg]);
	});
	debug(res);
	if (res) exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART2");
	return;
}

function main() {
	let data = fs.readFileSync("08_input").toString("utf-8");

	part1(data);
	part2(data);
	process.exit(0);
}

main();
